### DISCUSSION TITLE
public mode - private mode

### DISCUSSION THREAD

**Discussion by SimonTheWay** [danielnilsson9/bbs-fw#99] [Ideas]
Hi @danielnilsson9,

I really appreciate your work. Im thinking about the way to switch from private mode (full power) to public mode (street legal)

I'm inspired by my cooked firmware on my XIAOMI m365.

Normal start is in public mode, if you start the bike with brake pressed, it will switch in private mode

What do you think about this feature?

thanks for hearing me
Simon

**Comment by danielnilsson9** [comment]
Sounds like a convenient feature, feel free to open an issue and I will try to include it in next release.

**Comment by silenec** [comment]
I mean it sort of is already there - Standard vs Sport mode toggled by PAS0 + light button. Also less chance of pressing that accidentally than touching a brake when starting on a hill for example.

**Comment by Delphis1982** [comment]
The problem of this kind of combination is, that the local police is able to read out the config and then activate the offroad mode -> Busted.
It would be better to write a little config tool for configuring the offroad mode (ONLY the offroad mode).
You set the settings with the usual tool for street legal, but offroad is set with a separate tool.
That tool would not set some kind of config file that could be read out,
but it would completely compile the whole firmware to flash with the settings integrated inside the firmware.
So you could define a **configurable** (important) trigger (like a pin) for mode change
and you are not able to readout that settings. You have to read out the firmware, and AFAIK this is actually not possible. So you're safe.

I'm trying to figure out myself to integrate my offroad settings inside the source code but I'm not a programmer. It's learning by doing.
I simply don't want ANYONE being able to read ANYTHING of the offroad settings.
It shouldn't even be visible that there is a hidden setting at all.
So you can only be charged if the local authorities would find the right firmwarefile or the source code on your hard disk at home !!!

By the way.
There are many options under the changing trigger for the operation mode.
I know about Lights button, Brake sensors or PAS-Level,
but what is a "sports button" and how would I search for it and how could i connect it at the BBSHD?
The Sports button could be some key like a car key to activaste offroad mode.
If you are searched and no such key would be found, then they could read in the config what they want.
If I am not able to activate the illegal mode "on the fly" then there isn't any problem with it.

**Reply by WvBijsterveld** [reply]
i absolutely get what you're saying, creating a pin is possible you just have to try things as you said. but also how do you see the police trying to read the config file?
daniels code isn't the shipped code with the motor and you are assuming the police have a programming cable as well. if the police are that well educated that they have a programming cable and know you're using daniels software they would be smart enough to know you can compile own functions into it.
although they have no evidence they would know what to search for. but i personally think you're thinking too much about it.

the sports button is the sports mode on the dpc-18 screen and it's too obvious.

i recommend sticking to the pas(6 or any number) and lights on or carrying a brake sensor and using that at startup.
or get a eggrider and you can only turn sport mode on via your phone.
eggrider is really worth the money as you get range estimation, and a lot of info.
